### PR TITLE
Fixed jsonschema warnings.

### PR DESCRIPTION
**Pull request by RamosFe**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/20748|

## Description

Fixed `jsonschemma` deprecated warnings in the `validator.py` and `test_validator.py` files.

### CODE REVIEW

File: api/api/test/test_validator.py [hunk 1]
```diff
@@ -307,4 +307,4 @@ def test_check_component_configuration_pair(component, configuration, expected_r
         assert isinstance(response, expected_response.__class__)
         assert response.code == expected_response.code
     else:
-        assert response is expected_response
+        assert response is expected_response
```
Comment (GGP1) [inline api/api/test/test_validator.py:310]: Add EOF line. You can configure VSCode so they are added automatically -> https://stackoverflow.com/questions/44704968/visual-studio-code-insert-newline-at-the-end-of-files

File: api/api/validator.py [hunk 2]
```diff
@@ -340,155 +340,155 @@ def check_component_configuration_pair(component: str, configuration: str) -> Wa
...
+@Draft4Validator.FORMAT_CHECKER.checks("group_names")
 def format_group_names(value):
     return check_exp(value, _group_names)
 
 
-@draft4_format_checker.checks("group_names_or_all")
+@Draft4Validator.FORMAT_CHECKER.checks("group_names_or_all")
 def format_group_names_or_all(value):
-    return check_exp(value, _group_names_or_all)
+    return check_exp(value, _group_names_or_all)
```
Comment (GGP1) [inline api/api/validator.py:494]: Add EOF line

**Review by GGP1 — CHANGES_REQUESTED** [review]

File: api/api/test/test_validator.py (hunk 1 above)
Comment (RamosFe) [reply api/api/test/test_validator.py]: Fixed in 8e322dd450925748ac5864038edb1c5146488d11.

File: api/api/validator.py (hunk 2 above)
Comment (RamosFe) [reply api/api/validator.py]: Fixed in 8e322dd450925748ac5864038edb1c5146488d11.

**Review by GGP1 — APPROVED** [review]
LGTM, failed checks aren't caused by the changes introduced

**Review by Selutario — APPROVED** [review]

**Pull request merged**